crates/re_space_view_time_series/src/line_visualizer_system.rs

```diff
@@ -96,156 +96,193 @@ impl SeriesLineSystem {
+                                break;
+                            }
+                            if let Some(color) = color.first().copied().flatten().map(|c| {
+                                let [r,g,b,a] = c.to_array();
+                                if a == 255 {
+                                    // Common-case optimization
+                                    re_renderer::Color32::from_rgb(r, g, b)
+                                } else {
+                                    re_renderer::Color32::from_rgba_unmultiplied(r, g, b, a)
                                 }
+                            }) {
+                                points[i].attrs.color = color;
                             }
                         }
+                    }
+                }
 
-                        // Fill in radii -- if available _and_ not overridden.
-                        if override_stroke_width.is_none() {
-                            if let Some(stroke_widths) = stroke_widths {
-                                for (i, stroke_width) in stroke_widths.range(entry_range.clone()).enumerate() {
-                                    if i >= stroke_widths.num_entries() {
-                                        re_log::debug_once!("more stroke width attributes than points in {entity_path:?} -- this points to a bug in the query cache");
-                                        break;
-                                    }
-                                    if let Some(stroke_width) = stroke_width.first().copied().flatten().map(|r| r.0) {
-                                        points[i].attrs.marker_size = stroke_width;
-                                    }
-                                }
+                // Fill in radii -- if available _and_ not overridden.
+                if override_stroke_width.is_none() {
+                    if let Some(stroke_widths) = stroke_widths {
+                        for (i, stroke_width) in stroke_widths.range(entry_range.clone()).enumerate() {
+                            if i >= stroke_widths.num_entries() {
+                                re_log::debug_once!("more stroke width attributes than points in {entity_path:?} -- this points to a bug in the query cache");
+                                break;
+                            }
+                            if let Some(stroke_width) = stroke_width.first().copied().flatten().map(|r| r.0) {
+                                points[i].attrs.marker_size = stroke_width;
                             }
                         }
-                    },
-                )?;
-            }
-
-            // Check for an explicit label if any.
-            // We're using a separate latest-at query for this since the semantics for labels changing over time are a
-            // a bit unclear.
-            // Sidestepping the cache here shouldn't be a problem since we do so only once per entity.
```

Comment:
That seemed like a valuable comment?

Reply:
I agree - it's still in there when I check out the code, but GitHub shows is as gone O.o

Reply:
Weird! Rust-analyzers's "Extract to function" removed it. Good catch!